### PR TITLE
Dynamically get stories related to a Project

### PR DESCRIPTION
**Pull request by esirK**
## Description
Get and display stories related to a Project.

Finishes https://github.com/CodeForAfrica/ui/issues/190

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots
Screenshot 2022-08-11 at 12 34 42

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/codeforafrica/src/lib/api.ghost/index.js [hunk 1]
```diff
@@ -47,3 +47,10 @@ export async function getRelatedStories(story) {
       s.tags.some((t) => story.tags.find((st) => equalsIgnoreCase(t, st)))
   );
 }
+
+export async function getStoriesRelatedToProject(projectName) {
+  const stories = await getAllStories();
+  return stories.filter((s) =>
+    s.tags.some((t) => equalsIgnoreCase(t, projectName))
+  );
+}
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/api.ghost/index.js:56]: I think the two functions `getRelatedStories` and `getStoriesRelatedToProject` can be combined into `getRelatedStoriesByTags(...tags)` where tags is an array of strings.

For stories, we would then call this function `getRelatedStoriesByTags(story.tags)` and for projects, we could call it as `getRelatedStoriesByTags([project.name])`... or something like that.

**Review by kilemensi — APPROVED** [review]
🚀

File: apps/codeforafrica/src/lib/api.ghost/index.js [hunk 2]
```diff
@@ -35,15 +35,11 @@ export async function getStory(slug) {
   return getPost(slug);
 }
 
-export async function getRelatedStories(story) {
-  if (!story?.tags?.length) {
-    return null;
-  }
-
+export async function getRelatedStoriesByTags(tags, story = {}) {
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/api.ghost/index.js:38]:
```suggestion
export async function getRelatedStoriesByTags(tags) {
```

Why do we still need `story`?

File: apps/codeforafrica/src/lib/index.js [hunk 3]
```diff
@@ -387,12 +389,14 @@ async function getProjectPageStaticProps(params) {
...
-            articles: [],
+            articles: relatedStories.slice(0, 3),
           },
           {
             slug: "related-projects",
             title: "Explore other projects",
-            projects: projects.slice(0, 3),
+            projects: projects
+              .filter((p) => p.slug !== project.slug)
+              .slice(0, 3),
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/index.js:399]: :100:

File: apps/codeforafrica/src/lib/api.ghost/index.js (hunk 2 above)
Comment (esirK) [reply apps/codeforafrica/src/lib/api.ghost/index.js:38]: To prevent showing current story under `News and Stories`

Comment (kilemensi) [reply apps/codeforafrica/src/lib/api.ghost/index.js:38]: We can `filter`/`slice` the returned list similar to how we're doing for related projects @esirK .

**Pull request merged**